charts/templates/statefulset.yaml

```diff
@@ -137,11 +137,13 @@ spec:
                   - bash
                   - '-c'
                   - |
-                    pid=$(ps aux | grep -i '/bin/bas[h] /usr' | awk {'print $2'})
+                    pid=$(pgrep start.marklogic)
 
                     log () {
                       local TIMESTAMP=$(date +"%Y-%m-%d %T.%3N")
-                      echo "${TIMESTAMP}  $@" > /proc/$pid/fd/1
+                      if [ -n "$pid" ]; then
+                          echo "${TIMESTAMP} $@" > /proc/$pid/fd/1
+                      fi
```

Comment:
There should be a comment with this if block indicating why the check is done as this is a very unexpected case if the pid lookup code is working as it should.

Reply:
Is there somewhere else we can report the error that we can't find a pid to log a message too? Seems like a catch 22.

Reply:
I can add the comment regarding the condition check. But for logging the empty pid message i don't think we have option to report it somewhere else other than forwarding them to files. If anything is possible then we would have not used PID in first place to report the log messages. Will create ticket for this as well post GA to see if we can can report logs to some common location.

Reply:
Added comment